### PR TITLE
feat: initial wallet service implementation

### PR DESCRIPTION
**Pull request by alexanderwende**
wallet service implementation;
example page setup;
repository setup;

### CODE REVIEW

File: src/connection/providers/browser/provider.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+        // ensure the browser wallet identifier is stored in the connection options
+        // connectionOptions.walletId = getProviderId(wallet);
+        connectionOptions.walletIdentifier = getEthereumProviderIdentifier(wallet);
+
+        // check if we have a cached connection matching the current connection options
+        const cachedConnection = this.getCachedConnection(connectionOptions);
+
+        // if we do, return our cached connection instead of connecting again
+        if (cachedConnection) {
```
Comment (JTraversa) [inline src/connection/providers/browser/provider.ts:71]: 👍  Provider connection looks good

File: src/connection/providers/browser/ethereum-provider/eip-6963.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    return providers;
+}
+
+/**
+ * Get the EIP-6963 provider (browser wallet) that matches the given identifier.
+ *
+ * @remarks
+ * On page reload, the `uuid`s of the EIP-6963 providers change, meaning we cannot use them to identify wallets
+ * over multiple sessions. To alleviate this, we use the `name` and `rdns` properties to provide a 'fuzzy' match
```
Comment (JTraversa) [inline src/connection/providers/browser/ethereum-provider/eip-6963.ts:18]: This is interesting that the uuid's change? Is there a rationale youve understood for why? Wouldnt a wallet just have a consistent universal id?

Comment (JTraversa) [reply src/connection/providers/browser/ethereum-provider/eip-6963.ts:18]: Also 👍 for the fuzzy workaround.

I think one thing to keep in mind is that really if someone installs a malicious extension theyre likely just screwed in so many ways already as well

**Review by JTraversa — APPROVED** [review]
LGTM! I mostly went through to make sure I understood the implementation and its implications. 

Beyond understanding the EIP stuff, I also did a review of the utils, and most of the light files and they were all 👍 as well

File: src/connection/providers/browser/ethereum-provider/eip-6963.ts (hunk 2 above)
Comment (alexanderwende) [reply src/connection/providers/browser/ethereum-provider/eip-6963.ts:18]: > This is interesting that the uuid's change? Is there a rational youve understood for why? Wouldnt a wallet just have a consistent universal id?

I haven't found a specific rationale for this, but basically the EIP-6963 spec states, that provider uuid's actually identify "provider sessions [...] during the lifetime of the page.". So my best guess is, that there's no mechanism/service for browser wallet extensions to register for a globally unique id, so they scoped this down to a session id that's generated on the fly. Not _super_ useful really...

**Pull request merged**